File: Windows/cpp/workspaces/warehouse/src/core/Picker.cpp
#include "Picker.h"
#include <algorithm>

namespace core {
    int Picker::Run(Warehouse& warehouse, int id, int attempts) {
        int successes = 0;
        for (int i = 0; i < attempts; ++i) {
            if (warehouse.Reserve(id, 1)) {
                ++successes;
            }
        }
        return successes;
    }
} // namespace core

File: Windows/dotnet4-x/workspaces/warehouse/src/Core/Picker.cs
using System;
using System.Threading;

namespace App
{
    class Picker
    {
        public int Run(Warehouse warehouse, int id, int attempts)
        {
            int successCount = 0;
            for (int i = 0; i < attempts; i++)
            {
                if (warehouse.Reserve(id, 1))
                {
                    successCount++;
                }
            }
            return successCount;
        }
    }
}